nexus/db-queries/src/db/datastore/fm.rs

```diff
@@ -139,10 +151,181 @@ impl DataStore {
+                                .into(),
+                            comment: assignment.comment,
+                        })
+                        .map_err(|_| {
+                            let internal_message = format!(
+                                "encountered multiple case ereports for case \
+                                 {case_id} with the same UUID {id}. this \
+                                 should really not be possible, as the \
+                                 assignment UUID is a primary key!",
+                            );
+                            Error::InternalError { internal_message }
+                        })?;
+                }
+            }
+
+            map
+        };
+        // Next, load the case metadata entries and marry them to the sets of
+        // ereports assigned to those cases that we loaded in the previous step.
+        let cases = {
+            let mut cases = iddqd::IdOrdMap::new();
+            let mut paginator =
+                Paginator::new(SQL_BATCH_SIZE, PaginationOrder::Descending);
+            while let Some(p) = paginator.next() {
+                let batch = self
+                    .fm_sitrep_cases_list_on_conn(
+                        id,
+                        &p.current_pagparams(),
+                        &conn,
+                    )
+                    .await
+                    .map_err(|e| {
+                        e.internal_context("failed to list sitrep cases")
+                    })?;
+                paginator = p.found_batch(&batch, &|case| case.id);
+                cases.extend(batch.into_iter().map(|case| {
+                    let model::fm::CaseMetadata {
+                        id,
+                        sitrep_id: _,
+                        created_sitrep_id,
+                        closed_sitrep_id,
+                        comment,
+                        de,
+                    } = case;
+                    let id = id.into();
+
+                    // Take all the case ereport assignments we've collected for this case.
+                    let ereports = case_ereports
+                        .remove(&id)
+                        // If there's no entry in the map of case ereport
+                        // assignments for this case, then that just means that the
+                        // case has no ereports assigned to it, so insert an empty
+                        // map here.
+                        .unwrap_or_default();
+                    fm::Case {
+                        id,
+                        created_sitrep_id: created_sitrep_id.into(),
+                        closed_sitrep_id: closed_sitrep_id.map(Into::into),
+                        de: de.into(),
+                        comment,
+                        ereports,
+                    }
+                }));
+            }
 
-        Ok(Sitrep { metadata })
+            cases
+        };
+
+        Ok(Sitrep { metadata, cases })
```

Comment:
Nice, this loading structure seems like it'll scale for the forseeable future. Thanks for making these changes!